src/Compilers/Core/Portable/SourceGeneration/GeneratedCodeUtilities.cs

```diff
@@ -159,18 +159,27 @@ internal static GeneratedKind GetIsGeneratedCodeFromOptions(ImmutableDictionary<
             return GeneratedKind.Unknown;
         }
 
-        internal static bool? GetIsGeneratedCodeFromOptions(AnalyzerConfigOptions options)
+        internal static GeneratedKind GetGeneratedCodeKindFromOptions(AnalyzerConfigOptions options)
```

Comment:
needed in this PR?

Reply:
Yes.

Reply:
We need GetIsGeneratedCodeFromOptions overload that takes AnalyzerConfigOptions